lib/train/transports/ssh.rb

```diff
@@ -85,6 +88,14 @@ def connection(state = {}, &block)
 
     private
 
+    def reusable_connection?(conn_opts)
+      return false unless @connection_options
+
+      current_options = @connection_options.dup
+
+      current_options.all? { |k, v| conn_opts[k] == v }
```

Comment:
Drive-by comment: this looks like a bug to me. If the intended behavior is to test that `@connection_options` is a subset of `conn_opts` and not that they are equal, a comment would be a nice way to indicate that to future code readers.